QUARKUS-4958.md

```diff
@@ -0,0 +1,62 @@
+# QUARKUS-4958 - Update Spring APIs to Spring Boot 3
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-4958
+
+Upstream PRs:
+- https://github.com/quarkusio/quarkus/pull/40344
+- https://github.com/quarkusio/quarkus/pull/41206
+- https://github.com/quarkusio/quarkus/pull/41993
+
+Spring Data and Spring Data REST [has been updated](https://github.com/quarkusio/quarkus-spring-data-api/pull/9)
+to the 3.2.5 and 4.2.5 respectively.
+Quarkus supports Spring interfaces only in documented use cases.
+That is, this test plan is only interested in a Spring API changes supported by Quarkus.
+Following changes are relevant:
+- [New CRUD repository interfaces that return List instead of Iterable](https://github.com/spring-projects/spring-data-commons/wiki/Spring-Data-2022.0-(Turing)-Release-Notes#new-crud-repository-interfaces-that-return-list-instead-of-iterable)
+- [Sorting repositories no longer inherit from CRUD repositories](https://github.com/spring-projects/spring-data-commons/wiki/Spring-Data-2022.0-(Turing)-Release-Notes#sorting-repositories-no-longer-inherit-from-crud-repositories)
+- [JpaRepository.getReferenceById replaces JpaRepository.getById](https://github.com/spring-projects/spring-data-commons/wiki/Spring-Data-2021.2-(Raj)-Release-Notes#jparepositorygetreferencebyid-replaces-jparepositorygetbyid)
+
+We are not really interested in other Spring changes and features (as support for them is not documented).
+Breaking changes goes mostly down to the interface hierarchy changes and excessive refactoring inside Quarkus Spring Data JPA deployment module.
+
+## Scope of the testing
+Following additions should be tested:
+- `getReferenceById` as replacement for deprecated `getOne` and `getById`
+- `ListCrudRepository` and `ListPagingAndSortingRepository` new interfaces
+- test above-mentioned with `@Transactional` as that is missing in our existing test coverage
+- test `Sort` and `TypedSort` in native as newly, [there is related native image substitution](https://github.com/quarkusio/quarkus/blob/main/extensions/spring-data-jpa/runtime/src/main/java/io/quarkus/spring/data/runtime/Target_org_springframework_data_domain_Sort_TypedSort.java)
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Focus on exploratory testing of the feature
+- Ensure good user experience and simplicity of use
+
+## Existing test coverage
+Quarkus QE Test Suite has existing test coverage that was affected by these breaking changes.
```

Comment:
Also we have some coverage in https://github.com/quarkus-qe/beefy-scenarios/tree/main/602-spring-data-rest but from I remember it's same or similar to TS coverage.

Reply:
I didn't realize that, good catch. IMO it's not relevant because this TP is concerned with testing RHBQ and we don't run beefy scenarios with a product.

Reply:
@jedla97 it's like mentioning FW examples, they are there, but I don't take them into consideration. Anyway, I didn't realize them, otherwise I'd inspect beefy scenarios.